### PR TITLE
feat: login with google button created

### PR DESCRIPTION
**Pull request by sumanjeet0012**
## Related Issue

fix: #139 
## Description

The login with google button is created and working properly but after login when user is redirected to /feed page then the middleware sends user back to the login page.

I don't know how to solve it.
I need some suggestions on how to solve it.

## Screenshots

image

https://github.com/Sanchitbajaj02/palettegram/assets/78401409/fc3183b3-83c7-40a7-bbfe-bc07581af898

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "react": "18.2.0",
     "react-dom": "18.2.0",
     "react-feather": "^2.0.10",
+    "react-icons": "^4.12.0",
```
Comment (Sanchitbajaj02) [inline package.json:35]: remove this library since it's too big

File: src/components/pages/auth/login/index.tsx [hunk 2]
```diff
@@ -11,9 +11,11 @@ import { toastify } from "@/helper/toastify";
 
 // API
 import { loginUser } from "@/backend/auth.api";
+import { account } from "@/backend/appwrite.config";
```
Comment (Sanchitbajaj02) [inline src/components/pages/auth/login/index.tsx:14]: configure the appwrite related stuff in the backend folder

File: src/components/pages/auth/login/index.tsx [hunk 3]
```diff
@@ -176,6 +178,24 @@ export default function LoginComponent() {
               </button>
             </div>
           </form>
+          <div>
```
Comment (Sanchitbajaj02) [inline src/components/pages/auth/login/index.tsx:181]: UI is not in sync with the existing one, please update

**Review by Sanchitbajaj02 — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (sumanjeet0012) [reply package.json:35]: The google icon is not present in "react-feather" , that's why I have used react-icons.
I will remove the "react-icons" .

File: src/components/pages/auth/login/index.tsx (hunk 2 above)
Comment (sumanjeet0012) [reply src/components/pages/auth/login/index.tsx:14]: okay I will create a function in backend folder and import it here.

File: src/components/pages/auth/login/index.tsx (hunk 3 above)
Comment (sumanjeet0012) [reply src/components/pages/auth/login/index.tsx:181]: okay I will do that.

**Comment by sumanjeet0012** [comment]
I will change all the things you have pointed out, but I need some help on the "Login with google" button . 

The button is working as expected and after success it redirects user to feeds page and may be due to middleware user again gets redirected to login page.

I need help in how the cookies will get created after logging in via "Login with google" button , so that the user can access the feed page..

**Comment by Sanchitbajaj02** [comment]
> I will change all the things you have pointed out, but I need some help on the "Login with google" button .
> 
> The button is working as expected and after success it redirects user to feeds page and may be due to middleware user again gets redirected to login page.
> 
> I need help in how the cookies will get created after logging in via "Login with google" button , so that the user can access the feed page..

Currently the validation is that their account id is storing in the cookies and if that is present as well as if the user's isVerified flag is true, then only user can visit the feed page

File: src/components/pages/auth/login/index.tsx (hunk 2 above)
Comment (sumanjeet0012) [reply src/components/pages/auth/login/index.tsx]: Done, Moved it to backend folder

File: package.json (hunk 1 above)
Comment (sumanjeet0012) [reply package.json]: Removed the react-icons

File: src/components/pages/auth/login/index.tsx (hunk 3 above)
Comment (sumanjeet0012) [reply src/components/pages/auth/login/index.tsx]: I have placed the login with google button inside the form.
Is it ok, or something else should be done here

**Review by Sanchitbajaj02 — APPROVED** [review]

**Comment by Sanchitbajaj02** [comment]
@sumanjeet0012 Your PR has been reviewed and approved. Great work 👏. 

Please resolve the merge conflicts by taking a pull from master and again push it back. I will merge it by EOD.

**Pull request merged**